plotter/scatter.go

```diff
@@ -24,22 +28,32 @@ type Scatter struct {
 // NewScatter returns a Scatter that uses the
 // default glyph style.
 func NewScatter(xys XYer) (*Scatter, error) {
+	var s Scatter
 	data, err := CopyXYs(xys)
 	if err != nil {
 		return nil, err
 	}
-	return &Scatter{
-		XYs:        data,
-		GlyphStyle: DefaultGlyphStyle,
-	}, err
+	s.XYs = data
+
+	s.GlyphStyle = DefaultGlyphStyle
+
+	s.GlyphStyleFunc = func(int) draw.GlyphStyle {
+		return s.GlyphStyle
+	}
+	return &s, err
 }
 
 // Plot draws the Scatter, implementing the plot.Plotter
 // interface.
 func (pts *Scatter) Plot(c draw.Canvas, plt *plot.Plot) {
 	trX, trY := plt.Transforms(&c)
-	for _, p := range pts.XYs {
-		c.DrawGlyph(pts.GlyphStyle, vg.Point{X: trX(p.X), Y: trY(p.Y)})
+	for i, p := range pts.XYs {
+		if pts.GlyphStyleFunc != nil {
```

Comment:
instead of testing over and over whether `GlyphStyleFunc` is nil or not, couldn't we create a little closure (before the for-loop) that will return the correct `GlyphStyle` value?

something like:
```go
glyph := func(i int) draw.GlyphStyle { return pts.GlyphStyle }
if pts.GlyphStyleFunc != nil {
    glyph = pts.GlyphStyleFunc
}
for i, p := range pts.XYs {
    c.DrawGlyph(glyph(i), vg.Point{X: trX(p.X), Y: trY(p.Y)})
}
```

Reply:
I do absolutely agree with you, have made the changes.

As for a separate color scatter plot, I am happy to bring it back, as it looks much better on its own, than together with 3 others.